### PR TITLE
Update fish to 3.5.0 to resolve CVE-2022-20001

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Fish is a new package, updating to the latest version to resolve a CVE rather than patching since it is so new.

Note: I tried to bump to the latest RHEL spec but it is fairly incompatible with our current environment. Went with piecemeal updates to the spec instead.

###### Change Log  

- Update Fish to 3.5.0 to resolve CVE-2022-20001
- Add a runtime requirement on `util-linux` to fix an error in Fish

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-20001

###### Test Methodology

- Local build and testing.
- Waiting on check build

### CODE REVIEW

File: SPECS/fish/fish.spec [hunk 1]
```diff
@@ -28,7 +28,7 @@ Distribution:   Mariner
 #   - user_doc/html/_static/underscore.js
 License:        GPLv2 and BSD and ISC and LGPLv2+ and MIT
 URL:            https://fishshell.com
-Source0:        https://github.com/fish-shell/fish-shell/releases/download/%{version}/%{name}-%{version}.tar.gz
+Source0:        https://github.com/fish-shell/fish-shell/releases/download/%{version}/%{name}-%{version}.tar.xz
```
Comment (PawelWMS) [inline SPECS/fish/fish.spec:31]: I think we're missing a signature update.

File: cgmanifest.json [hunk 2]
```diff
@@ -3188,8 +3188,8 @@
         "type": "other",
         "other": {
           "name": "fish",
-          "version": "3.1.2",
-          "downloadUrl": "https://github.com/fish-shell/fish-shell/releases/download/3.1.2/fish-3.1.2.tar.gz"
+          "version": "3.5.0",
```
Comment (jslobodzian) [inline cgmanifest.json:3191]: Are there any packages that depend on fish that need to be bumped as well...given the large version jump from 3.1 to 3.5.  (Theoretically this is ok...but just want to double check)

Comment (dmcilvaney) [reply cgmanifest.json:3191]: Updated two packages (one is in specs-extended which I don't have a system setup to build quickly, so untested).
gh builds fine with the new fish

File: SPECS/fish/fish.spec (hunk 1 above)
Comment (dmcilvaney) [reply SPECS/fish/fish.spec:31]: Oops, copied the updated file the wrong way... fixed now.

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**